src/OrchardCore/OrchardCore/Localization/BclCalendars.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using NodaTime;
+
+namespace OrchardCore.Localization
+{
+    internal static class BclCalendars
+    {
+        public static IEnumerable<Calendar> MappedCalendars =>
+            new[] { Hebrew, Hijri, Gregorian, Julian, Persian, UmAlQura };
+
+        public static Calendar Hebrew => new HebrewCalendar();
+        public static Calendar Hijri => new HijriCalendar();
+        public static Calendar Gregorian => new GregorianCalendar();
+        public static Calendar Julian => new JulianCalendar();
+        public static Calendar Persian => new PersianCalendar();
+        public static Calendar UmAlQura => new UmAlQuraCalendar();
+
+        public static CalendarSystem ConvertToCalendarSystem(Calendar calendar)
```

Comment:
We are using a string in the results to hide NodaTime types which is fine. But what about creating our own enumeration of calendars in the Abstractions so that we don't have to do a lookup and can just return the enumeration value in the CultureSelectResult. This way we prevent serialization and lookup, and keep this part of the code in the method that needs the CalendarSystem type.